starknet_py/contract.py

```diff
@@ -381,10 +395,71 @@ async def from_address(
         :raises BadRequest: when contract is not found
         :param address: Contract's address
         :param client: Client used
+        :param resolve_proxies: Whether should `from_address` resolve methods from proxied contract or not
+        :param proxy_checks: List of classes implementing :class:`starknet_py.proxy_check.ProxyCheck` ABC,
+            that will be used for checking if contract at the address is a proxy contract. If no proxy_checks
+            are provided and ``resolve_proxies=True``,will default to :class:`starknet_py.proxy_check.ArgentProxyCheck`
+            and :class:`starknet_py.proxy_check.OpenZeppelinProxyCheck`
         :return: an initialized Contract instance
         """
+        if proxy_checks is None:
+            proxy_checks = []
+        proxy_checks += [ArgentProxyCheck(), OpenZeppelinProxyCheck()]
+
+        contract = await Contract._from_address(
+            address=address,
+            client=client,
+            proxy_checks=proxy_checks if resolve_proxies else [],
+            max_steps=max_steps,
+            step=1,
+        )
+
+        return Contract(address=address, abi=contract.data.abi, client=client)
+
+    @staticmethod
+    async def _from_address(
```

Comment:
it's confusing to have 2 different methods - one public and one private named the same